crates/datasources/Cargo.toml

```diff
@@ -6,6 +6,7 @@ edition = {workspace = true}
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+apache-avro = "0.15"
```

Comment:
It exports a wrapper around an avro reader for reading into record batches. Datafusion is also on 0.14 right now.

Using the avro stuff directly is scoped to reading the metadata, and allows us to use serde on the types. I don't think there will be a chance for conflict/confusion since avro isn't used outside the `spec` module (which itself doesn't depend on any datafusion stuff).